### PR TITLE
[CI] Add a CI job to run type checking with mypy

### PR DESCRIPTION
**Pull request by odesenfans**
Fixed type checking issues on the whole codebase.

### CODE REVIEW

File: .github/workflows/pyaleph-ci.yml [hunk 1]
```diff
@@ -8,6 +8,7 @@ on:
     branches:
       - dev
       - master
+      - od-mypy
```
Comment (hoh) [inline .github/workflows/pyaleph-ci.yml:11]: Really ?

File: mypy.ini [hunk 2]
```diff
@@ -0,0 +1,65 @@
+# Global options:
```
Comment (hoh) [inline mypy.ini:1]: Were all of these necessary for the codebase to pass ?

Comment (odesenfans) [reply mypy.ini:1]: No, these are a few default options I like. We can tweak it as we go.

File: .github/workflows/pyaleph-ci.yml (hunk 1 above)
Comment (odesenfans) [reply .github/workflows/pyaleph-ci.yml]: Fixed.

File: mypy.ini (hunk 2 above)
Comment (hoh) [reply mypy.ini:1]: OK.

**Pull request merged**